plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx

```diff
@@ -36,11 +36,14 @@ export const EntityPicker = ({
 }: FieldProps<string>) => {
   const allowedKinds = uiSchema['ui:options']?.allowedKinds as string[];
   const defaultKind = uiSchema['ui:options']?.defaultKind as string | undefined;
+  const allowedTypes = uiSchema['ui:options']?.allowedTypes as string[];
   const catalogApi = useApi(catalogApiRef);
 
   const { value: entities, loading } = useAsync(() =>
     catalogApi.getEntities(
-      allowedKinds ? { filter: { kind: allowedKinds } } : undefined,
+      allowedKinds
+        ? { filter: { kind: allowedKinds, type: allowedTypes } }
```

Comment:
Thanks for your comment, and sorry for the late reply. I just gave this a try and you're right, this does not work. However `spec.type` gave me a syntax error, probably due to the `.`. I'm currently thinking of modifying the CatalogEntitiesRequest to be able to accept object, but I'm not sure if this is the best course of action. Do you have any suggestions on this?

https://github.com/backstage/backstage/blob/5bdaccc40b4a814cf0b45d429f15a3afacc2f60b/packages/catalog-client/src/types/api.ts#L26-L37

Reply:
Well you are able to quote the keys. `{ 'some.key': 'some-value' }`

Still, I'm not entirely sure about the solution overall. Each kind has a different set of possible types, so having these two arrays side by side don't make a lot of sense. Also, preferably, the flow should be server data driven - the user might select a kind first, and then, based on what types that _actually exist_ on the server side, it could present a dropdown of the types to select among. And the type picker might not even appear if the kind in question doesn't have types to begin with.

I feel that to solve this right, the solution might be a bit more complex.

Reply:
Thank you for your suggestion @freben, quoting the keys works.

The reasoning behind this approach is that the maintainer of the entity catalog would have visibility on the available types for each kinds, and thus if the maintainer put in an invalid type in the filter, it should be treated similarly as if they put in an invalid kind in the allowedKinds filter. We do not do backend validation for kinds, so I'm not sure what improvements would doing backend validation for types bring over this approach.

Btw, I noticed that the PR has been closed. Do you have a guideline on reopening the PR, or should I create a new PR? We can continue this discussion in the new PR if a new one is created

Reply:
You should be able to re-open this PR and carry on with the work here I believe. Let me know if there's some problems and I can re-open it for you.

Reply:
@benjdlambert I can't re-open this PR. Can you help to reopen it? Thanks

Reply:
Done! :)

Reply:
Thanks! I've pushed the latest update that worked. Would love to hear comments and suggestions on this PR